funsor/adjoint.py

```diff
@@ -49,15 +49,7 @@ def interpret(self, cls, *args):
             self.tape.append((result, cls, args))
         else:
             result = self._old_interpretation.interpret(cls, *args)
-        lazy_args = [
-            self._eager_to_lazy.get(
-                id(arg)
-                if ops.is_numeric_array(arg) or not isinstance(arg, Hashable)
-                else arg,
-                arg,
-            )
-            for arg in args
-        ]
+        lazy_args = [self._eager_to_lazy.get(arg, arg) for arg in args]
```

Comment:
this was causing recursion errors because `.get()` calls `.__eq__` which calls `ops.equal` which creates a funsor etc. i'm unsure how to make this safer